libs/ngx.ux/src/viewport/viewport.service.ts

```diff
@@ -16,14 +16,29 @@ import { ViewportSizeTypeInfo, ViewportSize } from "./viewport.model";
 import { WindowRef } from "../platform/window";
 import { ViewportServerSizeService } from "./viewport-server-size.service";
 import { generateViewportSizeTypeInfoList, generateViewportSizeTypeInfoRefs, getSizeTypeInfo } from "./viewport.util";
-import type { Dictionary } from "../internal/internal.model";
 import { VIEWPORT_OPTIONS } from "./viewport.options";
 
 @Injectable({
 	providedIn: "root",
 })
 export class ViewportService {
 
+	readonly #viewportServerSize = inject(ViewportServerSizeService);
+	readonly #windowRef = inject(WindowRef);
+	readonly #config = inject(VIEWPORT_OPTIONS);
+
+	/** Viewport size types list ordered by type, smallest to largest. */
+	readonly sizeTypes = generateViewportSizeTypeInfoList(this.#config.breakpoints);
+
+	/** Size types refs of the generated viewport size type info. */
+	readonly sizeTypeMap = generateViewportSizeTypeInfoRefs(this.sizeTypes);
+
+	/** Viewport size signal (which is also throttled). */
+	readonly viewportSize = signal<ViewportSize>(this.#viewportServerSize.get())
```

Comment:
might not make any much diff but would start this as `undefined` or let it be initialized by https://github.com/sketch7/ssv.ngx/pull/19/files#r2111207000